### PR TITLE
Added an option to dispatch effects based on travelled distance

### PR DESCRIPTION
**Pull request by Reguas**
https://www.twitch.tv/videos/1803745281
https://www.twitch.tv/videos/1804685667
Meta effects are disabled in this mode; Twitch voting works fine

### CODE REVIEW

File: ChaosMod/Components/EffectDispatcher.h [hunk 1]
```diff
@@ -64,6 +64,12 @@ class EffectDispatcher : public Component
 		}
 	};
 
+	enum TravelledDistanceType
```
Comment (pongo1231) [inline ChaosMod/Components/EffectDispatcher.h:67]: Would be cleaner to use enum class here to not pollute the global namespace

File: ChaosMod/Components/EffectDispatcher.h [hunk 2]
```diff
@@ -95,6 +101,13 @@ class EffectDispatcher : public Component
 	std::list<RegisteredEffect *> m_DispatchedEffectsLog;
 
   private:
+	bool m_EnableDistanceBasedEffectDispatch = false;
+
+	float m_DistanceToActivateEffect = 500.f;
+	Vector3 m_SavedPosition = Vector3();
```
Comment (pongo1231) [inline ChaosMod/Components/EffectDispatcher.h:107]:
```suggestion
	Vector3 m_SavedPosition;
```

**Review by pongo1231 — CHANGES_REQUESTED** [review]

File: ChaosMod/Components/EffectDispatcher.h (hunk 1 above)
Comment (Reguas) [reply ChaosMod/Components/EffectDispatcher.h]: 👌

File: ChaosMod/Components/EffectDispatcher.cpp [hunk 3]
```diff
@@ -226,6 +226,11 @@ EffectDispatcher::EffectDispatcher(const std::array<BYTE, 3> &timerColor, const
 	m_MetaEffectShortDur =
 	    g_OptionsManager.GetConfigValue<int>("MetaShortEffectDur", OPTION_DEFAULT_EFFECT_META_SHORT_TIMED_DUR);
 
+	m_EnableDistanceBasedEffectDispatch = g_OptionsManager.GetConfigValue<bool>("EnableDistanceBasedEffectDispatch", false);
+	m_DistanceToActivateEffect = g_OptionsManager.GetConfigValue<int>("DistanceToActivateEffect", 250);
+	m_DistanceType                      = static_cast<TravelledDistanceType>(
+        g_OptionsManager.GetConfigValue<int>("DistanceType", static_cast<int>(TravelledDistanceType::Distance)));
```
Comment (pongo1231) [inline ChaosMod/Components/EffectDispatcher.cpp:232]: The default value should be a def under Util/OptionsDefault.h for consistency with the other GetConfigValue calls

**Review by pongo1231 — CHANGES_REQUESTED** [review]
Other than one thing LGTM

File: ChaosMod/Components/EffectDispatcher.cpp (hunk 3 above)
Comment (Reguas) [reply ChaosMod/Components/EffectDispatcher.cpp:232]: 👌

**Pull request merged**